src/Tizen.Peripheral/Tizen.Peripheral/Adc.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeAdc = Interop.Peripheral.Adc;
+
+namespace Tizen.Peripheral
+{
+    /// <summary>
+    /// The class allows applications to use the platform ADC peripheral.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Adc : IDisposable
+    {
+        /// <summary>
+        /// Native handle to ADC.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private Adc(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens the ADC pin.
+        /// </summary>
+        /// <param name="device">The ADC device number.</param>
+        /// <param name="channel">The ADC channel number to control.</param>
+        /// <returns></returns>
+        public static Adc Open(int device, int channel)
+        {
+            var ret = NativeAdc.Open(device, channel, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new Adc(handle);
+        }
+
+        /// <summary>
+        /// Gets the current value of the ADC pin.
+        /// </summary>
+        public uint Value
+        {
+            get
+            {
+                var ret = NativeAdc.Read(_handle, out uint value);
+                if (ret != Internals.Errors.ErrorCode.None)
+                    throw ExceptionFactory.CreateException(ret);
+
+                return value;
+            }
+        }
+
+        /// <summary>
+        /// Closes the ADC pin.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeAdc.Close(_handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes the ADC pin.
+        /// </summary>
+        public void Dispose()
```

Comment:
Implement the dispose pattern
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose